geniza/corpus/models.py

```diff
@@ -1293,8 +1293,35 @@ def merge_with(self, merge_docs, rationale, user=None):
         needs_review = [self.needs_review] if self.needs_review else []
 
         for doc in merge_docs:
+            # handle document dates validation before making any changes;
+            # mismatch should result in exception (caught by DocumentMerge.form_valid)
+            if (
+                doc.doc_date_standard
+                and self.doc_date_standard
+                and self.doc_date_standard != doc.doc_date_standard
+            ) or (
+                doc.doc_date_original
+                and self.doc_date_original
+                and self.doc_date_original != doc.doc_date_original
+            ):
+                raise ValidationError(
+                    "Merged documents must not contain conflicting dates; resolve before merge"
```

Comment:
Is it possible for one document to have a standard date and the other to have an original date that do not match each other? (seems unlikely, but possible?) Should that be considered conflicting and merit review?

Reply:
Good call, I think it makes sense to flag these to users even if it technically won't result in any data loss and is rare enough.